test/conformance/runtime/readiness_probe_test.go

```diff
@@ -28,31 +28,61 @@ import (
 	v1a1test "knative.dev/serving/test/v1alpha1"
 )
 
+// testCases for table-driven testing.
+var testCases = []struct {
+	// name of the test case, which will be inserted in names of routes, configurations, etc.
+	// Use a short name here to avoid hitting the 63-character limit in names
+	// (e.g., "service-to-service-call-svc-cluster-local-uagkdshh-frkml-service" is too long.)
+	name string
+	// handler to be used for readiness probe in user container.
+	handler corev1.Handler
+}{
+	{
+		"httpGet",
+		corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/healthz",
+			},
+		},
+	},
+	{
+		"exec",
+		corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"/ko-app/runtime", "probe"},
+			},
+		},
+	},
+}
+
 func TestProbeRuntime(t *testing.T) {
 	t.Parallel()
 	cancel := logstream.Start(t)
 	defer cancel()
 
 	clients := test.Setup(t)
 
-	names := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   "runtime",
-	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
This will detach from the logstream, I'd move that in here as part of the change to `t.Run`